Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1

```diff
@@ -52,33 +52,110 @@ function Get-ConfigValue {
     }
 }
 
-function Invoke-SqlCommand{
+function Get-ServiceFromDiscovery {
+    param(
+        [String] $DiscoveryUrl = $(throw "Please specify the Discovery Service Uri"),
+        [String] $Name = $(throw "Please specify a service or application name"),
+        [String] $Version = $(throw "Please specify a service or application version")
+    )
+
+    $discoveryServiceQuery = "/Services?`$filter=ServiceName eq '$Name' and Version eq $Version&`$select=ServiceUrl"
+
+    $uri = "$DiscoveryUrl$discoveryServiceQuery"
+
+    try {
+        $response = Invoke-WebRequest -Uri $uri -Method GET -UseDefaultCredentials
+    }
+    catch {
```

Comment:
I think you may want to specify the type in this catch. If something else were to go wrong, this exception would be very obscure